tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -11,130 +11,221 @@
-                          "Current RSS memory consumption: " + std::to_string(test_cur_vmrss) + " KB.\n");
+                          "Reference RSS memory consumption: " + std::to_string(references[VMRSS]) + " KB.\n" +
+                          "Current RSS memory consumption: " + std::to_string(measures[VMRSS]) + " KB.\n");
 
-    if ((!Environment::Instance().getCollectResultsOnly()) && (test_cur_vmhwm > ref_vmhwm))
+    if ((!Environment::Instance().getCollectResultsOnly()) && (measures[VMHWM] > references[VMHWM]))
         return TestResult(TestStatus::TEST_FAILED,
                           "Test failed: HWM (peak of RSS) virtual memory consumption is greater than reference.\n"
-                          "Reference HWM of memory consumption: " + std::to_string(ref_vmhwm) + " KB.\n" +
-                          "Current HWM of memory consumption: " + std::to_string(test_cur_vmhwm) + " KB.\n");
+                          "Reference HWM of memory consumption: " + std::to_string(references[VMHWM]) + " KB.\n" +
+                          "Current HWM of memory consumption: " + std::to_string(measures[VMHWM]) + " KB.\n");
 
     return TestResult(TestStatus::TEST_OK, "");
 }
 
 TestResult
 test_infer_request_inference(const std::string &model_name, const std::string &model_path,
-                             const std::string &target_device,
-                             const long &ref_vmsize, const long &ref_vmpeak, const long &ref_vmrss,
-                             const long &ref_vmhwm) {
+                             const std::string &target_device, const std::array<long, MeasureValueMax> &references) {
     log_info("Inference of InferRequest from network: \"" << model_path
                                                           << "\" for device: \"" << target_device << "\"");
-    long vmsize_before_test = 0, vmrss_before_test = 0,
-            test_cur_vmsize = 0, test_cur_vmpeak = 0,
-            test_cur_vmrss = 0, test_cur_vmhwm = 0;
-    std::chrono::system_clock::time_point t_start, t_end;
-    std::chrono::duration<double> t_diff;
-
-    vmsize_before_test = (long) getVmSizeInKB();
-    vmrss_before_test = (long) getVmRSSInKB();
-
-    Core ie;
-    CNNNetwork cnnNetwork = ie.ReadNetwork(model_path);
-    ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, target_device);
-    InferRequest infer_request = exeNetwork.CreateInferRequest();
-
-    log_info_ref_mem_usage();
-
-    t_start = std::chrono::system_clock::now();
-    int seconds = 1;
-    do {
-        infer_request.Infer();
+
+    MemCheckPipeline memCheckPipeline;
+
+    log_info("Reference values of virtual memory consumption:");
+    log_info(memCheckPipeline.get_measures_headers_as_str());
+    log_info(references[VMRSS] << "\t\t" << references[VMHWM] << "\t\t" << references[VMSIZE] << "\t\t"
+                               << references[VMPEAK] << "\t\t" << references[THREADS]);
```

Comment:
Enumerating MeasureValue explicitly complicates adding new metrics. Instead, concatenate all values until MeasureValueMax in MeasureValue order. Consider adding utility under util::

Reply:
Since we aren't using all of measures from MeasureValue in tests, I think it's YAGNI.
Anyway, enums aren't iterable, so need to implement it through new class with additional logic, and it will require new iteration of refactoring

Reply:
Done in ffc77be12aad796d941b09f86e162b764f3b0487